src/components/vue-form-builder.vue

```diff
@@ -1190,8 +1143,65 @@ export default {
       this.addPageName = "";
       this.updateState();
     },
-    deletePage() {
-      this.config.splice(this.pageDelete, 1);
+    // This function is used to calculate the new index of the references
+    calcNewIndexFor(index, referencedBy) {
+      if (index > this.pageDelete) {
+        return index - 1;
+      }
+      if (index === this.pageDelete) {
+        throw new Error(
+          `${this.$t(
+            "Can not delete this page, it is referenced by"
+          )}: ${referencedBy}`
+        );
+      }
+      return index;
```

Comment:
Use a ternary operator on return
```suggestion
      if (index === this.pageDelete) {
        throw new Error(
          `${this.$t(
            "Can not delete this page, it is referenced by"
          )}: ${referencedBy}`
        );
      }
      return index > this.pageDelete ? index - 1 : index;
```